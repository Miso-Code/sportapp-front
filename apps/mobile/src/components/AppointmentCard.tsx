import React from 'react'
import dayjs from 'dayjs'
import { useTranslation } from 'react-i18next'
import { View, StyleSheet } from 'react-native'
import { Card, Text, Button } from 'react-native-paper'

interface AppointmentCardProps {
	date: string
	trainer: string
	type: string
	location: string
}

const AppointmentCard: React.FC<AppointmentCardProps> = ({
	date,
	trainer,
	type,
	location,
	...props
}) => {
	const { t } = useTranslation()
	return (
		<Card elevation={1} style={styles.card} {...props}>
			<Card.Content style={styles.cardContent}>
				<Text variant='labelMedium'>
					{dayjs(date).format('DD/MM/YYYY hh:mm A')}
				</Text>
				<Text variant='titleMedium' style={styles.cardTitle}>
					{trainer}
				</Text>
				<Text variant='bodyMedium'>
					Tipo: {t(`preference.types.${type}`)}
				</Text>

				<View style={styles.chipContainer}>
					{type === 'virtual' ? (
						<Button mode='contained' onPress={() => {}} disabled>
							Ir a la cita
						</Button>
					) : (
						<Text testID='location' variant='bodyMedium'>
							Ubicaci√≥n: {location}
						</Text>
					)}
				</View>
			</Card.Content>
		</Card>
	)
}

const styles = StyleSheet.create({
	card: {
		backgroundColor: '#f2f2f2',
		width: '100%'
	},
	cardTitle: {
		fontWeight: 'bold'
	},
	cardContent: {
		justifyContent: 'center',
		marginTop: 10
	},
	chipContainer: {
		alignItems: 'flex-start',
		justifyContent: 'space-between',
		flexDirection: 'row',
		marginTop: 10
	}
})

export default AppointmentCard
