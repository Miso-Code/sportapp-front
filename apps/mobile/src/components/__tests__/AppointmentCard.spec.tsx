import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'

import AppointmentCard from '../AppointmentCard'
import { Button } from 'react-native-paper'

jest.mock('dayjs')

describe('AppointmentCard', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		component = renderer.create(
			<AppointmentCard
				date={new Date().toISOString()}
				trainer='trainer'
				type='virtual'
				location='location'
			/>
		)
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render a button for virtual appointments and no location', () => {
		expect(component.root.findByType(Button)).toBeDefined()
		expect(
			component.root.findAllByProps({ testID: 'location' })
		).toHaveLength(0)
	})

	it('should render a location for in person appointments and no button', () => {
		component.update(
			<AppointmentCard
				date={new Date().toISOString()}
				trainer='trainer'
				type='inPerson'
				location='location'
			/>
		)

		expect(
			component.root.findAllByProps({ testID: 'location' })
		).toHaveLength(3) // react native stuff ¯\_(ツ)_/¯
		expect(component.root.findAllByType(Button)).toHaveLength(0)
	})
})
