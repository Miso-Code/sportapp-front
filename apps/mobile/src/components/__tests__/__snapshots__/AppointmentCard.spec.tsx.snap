// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentCard should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#f2f2f2",
      "borderRadius": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "width": "100%",
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        borderRadiusStyles={{}}
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingVertical": 16,
            },
            {
              "justifyContent": "center",
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "letterSpacing": 0.5,
                  "lineHeight": 16,
                },
                undefined,
              ],
            ]
          }
        >
          2024-05-04
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "fontWeight": "bold",
                },
              ],
            ]
          }
        >
          trainer
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                undefined,
              ],
            ]
          }
        >
          Tipo: 
          preference.types.virtual
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 10,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(28, 27, 31, 0.12)",
                "borderRadius": 20,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(28, 27, 31, 0.12)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": 64,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            {
                              "color": "rgba(28, 27, 31, 0.38)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Ir a la cita
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
